### PR TITLE
[ApproximateNearestNeighbors] Add notebook and remove exactNearestNeighborsJoin API from the ann class

### PR DESCRIPTION
**Pull request by lijinf2**

### CODE REVIEW

File: notebooks/approx-nearest-neighbors.ipynb [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+   "source": [
+    "dim = 2000\n",
+    "dtype = 'float32'\n",
+    "np.random.seed(1)\n",
+    "\n",
+    "# items\n",
+    "num_vecs = 100000\n",
+    "vec = np.random.randn(dim).reshape([1,dim])\n",
+    "arr = np.random.randn(num_vecs).reshape([num_vecs,1])\n",
+    "items = arr * vec\n",
```
Comment (eordentlich) [inline notebooks/approx-nearest-neighbors.ipynb:52]: We might want to have a 'better' selection of vectors.  I see this is also in the exact knn notebook but didn't notice before.   The vectors are all multiples of a single vectors.   Perhaps we can find a better choice of vectors for the next release.

File: python/src/spark_rapids_ml/knn.py [hunk 2]
```diff
@@ -1323,4 +1331,4 @@ def approxSimilarityJoin(
             where item_vector v1 is one of the k nearest neighbors of query_vector v2 and their distance is dist(v1, v2).
         """
 
-        return self.exactNearestNeighborsJoin(query_df, distCol)
```
Comment (eordentlich) [inline python/src/spark_rapids_ml/knn.py:1326]: good catch

**Review by eordentlich — APPROVED** [review]
👍

**Comment by lijinf2** [comment]
build

**Pull request merged**